### ISSUE TITLE
Get current sorted column id

### ISSUE THREAD

**Issue by clementdevos** [pbeshai/react-taco-table#26]
Hi, 

Is it possible in any way to get the current sorted column? 
We'd like to keep the sort column and order even on data change. 

Cheers, 

Lilion12

**Comment by clementdevos** [comment]
Or would it be possible to get a "onSortChanged" callback?

**Comment by clementdevos** [comment]
Hey @pbeshai, 

I have a weird behaviour with this feature. 
I can log the changes to the console without problem, however, i store the changes in an object in my redux store then when in the render() method i pass the sort values like this

```
const sortData = this.props.salesAnalysisData.sortData;
initialSortColumnId = {sortData.column}
```

My store changes without problems and on render i can log the new sortData from the store but the sort is stuck on the first passed sortData. 

To recap : 

My store contains an initial storeData on the first column, asc order
onSort, i store the sortData in the redux store 
on render, i fetch the sortData from the store then pass it back to the initialSortColumnId 

Shouldn't it work this way? 

Cheers mate!

**Comment by pbeshai** [comment]
So if I understand you correctly, you are expecting the table to re-read `initialSortColumnId` when props change?

My intention was that those are only used for initialization (similar to defaultValue on form inputs).

Let me know if I am understanding your problem:
1. You pass initial data into the table, already sorted on column **A**, **Ascending**. `initialSortColumnId` is **A**. This works as expected.
2. You click to sort by column **B**. This sorts the data and fires `onSort(B, true, sortedData)`
3. You store sortedData in your store, which then gets passed as props to to the table as `data`. I would expect the table to render the same way as if nothing changed. Is this not the case?

This is where I'm confused. You then change `initialSortColumnId` to match whatever came back from the `onSort` callback? initialSortColumnId is not read again by the component after construction. 

If you can expand on your use case I'd be happy to help. I think it may tie into #20.

**Comment by clementdevos** [comment]
Hey @pbeshai, 

Your reply makes perfect sense.

I checked my code again before replying, we did a small mistake with the column's id. 
Our app gives the possibility to display different KPIS and the column's id were "[KPI][code]" so when the data was changed, the id changed along. This is why the sort was lost on the column. 

Fixed the problem for the column id so when the data changes, our table keeps the sort on the column. 
However, the sort still inverts itself, i'll investigate on my side to see if the data is correct, will talk about this in #27
